cmd/ipfs/daemon.go

```diff
@@ -192,22 +192,23 @@ func daemonFunc(req cmds.Request, re cmds.ResponseEmitter) {
 	// let the user know we're going.
 	fmt.Printf("Initializing daemon...\n")
 
-	if managed, _, _ := req.Option(adjustFDLimitKwd).Bool(); managed {
+	managefd, _ := req.Options[adjustFDLimitKwd].(bool)
+	if managefd {
 		if err := utilmain.ManageFdLimit(); err != nil {
 			log.Errorf("setting file descriptor limit: %s", err)
 		}
 	}
 
-	ctx := req.InvocContext()
+	cctx := env.(*oldcmds.Context)
```

Comment:
@Stebalien 
This is one if I'm not mistaken.

Reply:
Ah. I was focusing on the `Options` casts. In this case, we *do* want to panic. It's a programmer error. In the `Options` cases, we don't (it just means that the option wasn't passed (or the option has the wrong type but checking that would be a pain and I'd like to fix that with something like https://github.com/ipfs/go-ipfs-cmds/issues/50).

However, you're right. We do occasionally return an error instead of panicing when casting `env`. We should always panic.